src/aap_eda/tasks/orchestrator.py

```diff
@@ -121,43 +134,195 @@ def _run_request(
     return True
 
 
-def _make_user_request(
-    process_parent_type: ProcessParentType,
-    id: int,
-    request_type: ActivationRequest,
-) -> None:
-    """Enqueue a task to manage the activation with the given id."""
-    requests_queue.push(process_parent_type, id, request_type)
-    job_id = _manage_process_job_id(process_parent_type, id)
-    unique_enqueue("activation", job_id, _manage, process_parent_type, id)
+class QueueNotFoundError(Exception):
+    """Raised when a queue is not found."""
+
+    ...
+
+
+class RequestDispatcher:
+    """Dispatch process requests to the activation manager.
+
+    Handles the dispatch of requests for the orchestrator,
+    enqueueing the RQ task in the right queue.
+    """
+
+    _queues = None
+
+    @classmethod
+    @property
+    def queues(cls):
+        if cls._queues is None:
+            cls._queues = cls.get_queues()
+        return cls._queues
+
+    @staticmethod
+    def dispatch(
+        process_parent_type: ProcessParentType,
+        process_parent_id: int,
+        request_type: Optional[ActivationRequest],
+    ):
+        job_id = _manage_process_job_id(process_parent_type, process_parent_id)
+
+        if request_type in [
+            ActivationRequest.START,
+            ActivationRequest.AUTO_START,
+        ]:
+            queue_name = RequestDispatcher.get_most_free_queue_name()
+        else:
+            queue_name = RequestDispatcher.get_queue_name_by_parent_id(
+                process_parent_type,
+                process_parent_id,
+            )
+
+            # If the queue is old or doesn't exist, we use a valid one
+            # to make sure the request is processed and the restart
+            # policy is respected
+            if (
+                not queue_name
+                or queue_name not in settings.RULEBOOK_WORKER_QUEUES
+            ):
+                queue_name = RequestDispatcher.get_most_free_queue_name()
+
+        unique_enqueue(
+            queue_name,
+            job_id,
+            _manage,
+            process_parent_type,
+            process_parent_id,
+        )
+
+    @staticmethod
+    def get_queues() -> list[Queue]:
+        """Return the list of RQ queues configured for podman multinode."""
+        queues = []
+        for queue_name in settings.RULEBOOK_WORKER_QUEUES:
+            try:
+                queues.append(get_queue(queue_name))
+            except KeyError:
+                LOGGER.exception(f"Queue {queue_name} not found")
+                raise QueueNotFoundError(
+                    f"Queue {queue_name} not found"
+                ) from None
 
+        return queues
 
+    @staticmethod
+    def get_most_free_queue_name() -> str:
```

Comment:
This function fails to load balance work across multiple workers if there are pending jobs.  It should count running and queued jobs for that worker.

Reply:
Replace this function with random.choice

Reply:
I don't think I should count jobs, the goal is to be aware of the running processes, not the jobs, that can be any of type and usually are very quick. The goal is to distribute rulebook processes, not the jobs to manage them. 

If I use random.choice I could schedule a new rulebook process in a node that has no capacity to run more rulebooks, that would be inefficient and bad UX if there is another node with capacity.

Reply:
Currently this heuristic is exactly backwards in many cases so anything else is better.

Reply:
We should use a single queue for distributing jobs.    So we should have one queue for the rulebook process starts and one queue per worker for the monitoring tasks.